rust/crates/libs/edlcodegen-tools/src/lib.rs

```diff
@@ -16,3 +20,110 @@ pub fn edlcodegen_path() -> PathBuf {
 pub fn flatc_path() -> PathBuf {
     exes_path().join("flatbuffers/flatc.exe")
 }
+
+const PROGRAM_FILES_X86: &str = "ProgramFiles(x86)";
+const DEFAULT_VCTOOLS_VERSION_FILE_PATH: &str =
+    "VC\\Auxiliary\\Build\\Microsoft.VCToolsVersion.default.txt";
+const MSVC_PATH: &str = "VC\\Tools\\MSVC";
+const VC_TOOLS_X64: &str = "Microsoft.VisualStudio.Component.VC.Tools.x86.x64";
+const ENCLAVE_LIB_X64_PATH: &str = "lib\\x64\\enclave";
+const UCRT_LIB_X64_PATH: &str = "ucrt_enclave\\x64\\ucrt.lib";
+
+const VC_TOOLS_ARM64: &str = "Microsoft.VisualStudio.Component.VC.Tools.ARM64";
+const ENCLAVE_LIB_ARM64_PATH: &str = "lib\\arm64\\enclave";
+const UCRT_LIB_ARM64_PATH: &str = "ucrt_enclave\\arm64\\ucrt.lib";
+
+fn get_architecture_sub_paths() -> (&'static str, &'static str, &'static str) {
+    match env::consts::ARCH {
+        "x86_64" => (VC_TOOLS_X64, ENCLAVE_LIB_X64_PATH, UCRT_LIB_X64_PATH),
+        "aarch64" => (VC_TOOLS_ARM64, ENCLAVE_LIB_ARM64_PATH, UCRT_LIB_ARM64_PATH),
+        _ => panic!("Unsupported architecture: {}", env::consts::ARCH),
+    }
+}
+
+fn get_sdk_lib_path() -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
I can't make a coherent statement about this - but it really feels like this should be part of .ps1/config files, not Rust code.

I understand that this is the expedient and self-contained way to get it running under `cargo build`, so I'll approve as-is.